deepchecks/nlp/checks/train_test_validation/text_embeddings_drift.py

```diff
@@ -0,0 +1,145 @@
+        Dimension reduction method to use for the check. Dimension reduction is used to reduce the number of
+        embeddings dimensions in order for the domain classifier to train more efficiently on the data.
+        The 2 supported methods are PCA and UMAP. While UMAP yields better results (especially visually), it is much
+        slower than PCA.
+        Supported values:
+        - 'auto' (default): Automatically choose the best method for the data. Uses UMAP if with_display is True,
+        otherwise uses PCA for a faster calculation. Doesn't use dimension reduction at all if the number of embeddings
+        dimensions is less than 30.
+        - 'pca': Use PCA for dimension reduction.
+        - 'umap': Use UMAP for dimension reduction.
+        - 'none': Don't use dimension reduction.
+    num_samples_in_display : int , default: 500
+        Number of samples to display in the check display scatter plot.
+    """
+
+    def __init__(
+            self,
+            sample_size: int = 10_000,
+            random_state: int = 42,
+            test_size: float = 0.3,
+            min_meaningful_drift_score: float = 0.05,
+            dimension_reduction_method: str = 'auto',
+            num_samples_in_display: int = 500,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.sample_size = sample_size
+        self.random_state = random_state
+        self.test_size = test_size
+        self.min_meaningful_drift_score = min_meaningful_drift_score
+        if dimension_reduction_method is None:
+            dimension_reduction_method = 'none'
+        if dimension_reduction_method.lower() not in ['auto', 'umap', 'pca', 'none']:
+            raise ValueError(f'dimension_reduction_method must be one of "auto", "umap", "pca" or "none". '
+                             f'Got {dimension_reduction_method} instead')
+        self.dimension_reduction_method = dimension_reduction_method.lower()
+        self.num_samples_in_display = num_samples_in_display
+
+    def run_logic(self, context: Context) -> CheckResult:
+        """Run check.
+
+        Returns
+        -------
+        CheckResult
+            value: dictionary containing the domain classifier auc and a dict of column name to its feature
+            importance as calculated for the domain classifier model.
+            display: distribution graph for each column for the columns most explaining the dataset difference,
+            comparing the train and test distributions.
+
+        Raises
+        ------
+        DeepchecksValueError
+            If the object is not a Dataset or DataFrame instance
+        """
+        train_dataset = context.train
+        test_dataset = context.test
+
+        sample_size = min(self.sample_size, train_dataset.n_samples, test_dataset.n_samples)
+        headnote = """
+        <span>
+        The shown features are the features that are most important for the domain classifier - the
```

Comment:
First line shouldn't be here

Reply:
Done (whole thing was a left over copy+paste)